Automattic/pocket-casts-android
# Allow syncing of skip time settings

Fixes #303

## Description

This allows syncing of skip time settings _to the server_ for logged-in users. The app [supported syncing of skip time settings](https://github.com/Automattic/pocket-casts-android/blob/main/modules/services/servers/src/main/java/au/com/shiftyjelly/pocketcasts/servers/sync/SyncSettingsTask.kt#L15-L25) to the servers, but they were not synced as the `..NeedsSync` flags were not set for skip time settings. This PR sets those flags whenever skip time settings are changed. Those settings are synced to the server whenever syncing is triggered either by the user or after a pre-defined interval by the app.

Note that syncing _from the server_ worked fine which explains why the issue was resolved for those who updated the web player setting https://github.com/Automattic/pocket-casts-android/issues/303#issuecomment-1322919716.

P.S. For logged-out users, the app's preferences are auto-backed up by Google roughly every night when the device is idle according to their [backup schedule](https://developer.android.com/guide/topics/data/autobackup#BackupSchedule). If they change those settings after backup but before [restore](https://developer.android.com/guide/topics/data/autobackup#RestoreSchedule), which happens when the app is installed, either from the Play Store, or during device setup, they’re overridden with the last backup values. Backup and restore schedule in this scenario is controlled by Google, and looks like we cannot do much about it.

## Testing Instructions

1. Login to your `Pocket Casts` account on the web player and note down skip time settings. 
2. Install and launch the app.
3. Go to `Profile` -> `Settings` -> `General`.
4. Update `Skip forward time` and `Skip backward time` under player settings. 
5. Return to `Profile` home tab and tap on "Refresh now" at the bottom of the screen.
6. Check skip time settings on the web.
7. Notice that they're updated to values set in the Android app.
